tests/integration_tests/functional/test_api.py

```diff
@@ -724,30 +724,22 @@ def _drive_patch(test_microvm):
 
 def test_api_vsock(test_microvm_with_api):
     """Test vsock related API commands."""
-    if test_microvm_with_api.build_feature != 'vsock':
-        pytest.skip("This test is meant only for vsock builds.")
-
     test_microvm = test_microvm_with_api
     test_microvm.spawn()
     test_microvm.basic_config()
 
     response = test_microvm.vsock.put(
         vsock_id='vsock1',
-        guest_cid=15
-    )
-    assert test_microvm.api_session.is_status_no_content(response.status_code)
-
-    # Adding another vsock should be fine.
-    response = test_microvm.vsock.put(
-        vsock_id='vsock2',
-        guest_cid=16
+        guest_cid=15,
+        uds_path='vsock.sock'
     )
     assert test_microvm.api_session.is_status_no_content(response.status_code)
 
     # Updating an existing vsock is currently fine.
     response = test_microvm.vsock.put(
-        vsock_id='vsock2',
-        guest_cid=166
+        vsock_id='vsock1',
+        guest_cid=166,
```

Comment:
At the moment, device activation will just fail, since the driver will see an invalid CID. I'll look into a way of warning / rejecting this kind of invalid input.

Reply:
Discussed offline and decided this should be made part of a bigger task to apply some input validation at the time of the API request in a consistent manner for all components that apply (logger, kernel, vsock).